File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/APVerbatim.java
package LivinGrimoire;

import java.util.ArrayDeque;
import java.util.Arrays;
import java.util.List;

public class APVerbatim extends AlgPart {
    public ArrayDeque<String> sentences;

    public APVerbatim(String... sentences) {
        super();
        this.sentences = new ArrayDeque<>(Arrays.asList(sentences));
    }

    public APVerbatim(List<String> sentences) {
        super();
        this.sentences = new ArrayDeque<>(sentences);
    }

    @Override
    public String action(String ear, String skin, String eye) {
        return sentences.isEmpty() ? "" : sentences.pollFirst();
    }

    @Override
    public boolean completed() {
        return sentences.isEmpty();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/livinGrimoire/APVerbatim.kt
package livinGrimoire

import java.util.*

class APVerbatim : AlgPart {
    private var sentences: Queue<String> = ArrayDeque()

    constructor(vararg sentences: String) {
        this.sentences.addAll(Arrays.asList(*sentences))
    }

    constructor(list1: ArrayList<String>) {
        sentences = ArrayDeque(list1)
    }

    override fun action(ear: String, skin: String, eye: String): String {
        // Poll returns null if empty, so we return "" instead
        val sentence = sentences.poll()
        return sentence ?: ""
    }

    override fun completed(): Boolean {
        return sentences.isEmpty()
    }
}
